File: vibecheck-core/tests/fixtures/lru_cache/copilot.py
class Node:
    def __init__(self, key, value):
        self.key = key
        self.value = value
        self.prevNode = None
        self.nextNode = None

class LRUCache:
    def __init__(self, capacity):
        self.capacity = capacity
        self.cache_map = {}
        self.head = Node(0, 0)
        self.tail = Node(0, 0)
        self.head.nextNode = self.tail
        self.tail.prevNode = self.head

    def get(self, key):
        if key not in self.cache_map:
            return -1
        node = self.cache_map[key]
        self._move_to_front(node)
        return node.value

    def put(self, key, value):
        if key in self.cache_map:
            node = self.cache_map[key]
            node.value = value
            self._move_to_front(node)
            return
        if len(self.cache_map) >= self.capacity:
            self._evict_last()
        newNode = Node(key, value)
        self.cache_map[key] = newNode
        self._add_to_front(newNode)

    def _add_to_front(self, node):
        node.prevNode = self.head
        node.nextNode = self.head.nextNode
        self.head.nextNode.prevNode = node
        self.head.nextNode = node

    def _remove_node(self, node):
        prevNode = node.prevNode
        nextNode = node.nextNode
        prevNode.nextNode = nextNode
        nextNode.prevNode = prevNode

    def _move_to_front(self, node):
        self._remove_node(node)
        self._add_to_front(node)

    def _evict_last(self):
        lastNode = self.tail.prevNode
        self._remove_node(lastNode)
        del self.cache_map[lastNode.key]

    def get_size(self): return len(self.cache_map)

    def contains(self, key): return key in self.cache_map

    def clear_all(self):
        self.cache_map.clear()
        self.head.nextNode = self.tail
        self.tail.prevNode = self.head

    def peek_value(self, key):
        if key not in self.cache_map:
	        return -1
        return self.cache_map[key].value

    def get_keys(self):
        resultKeys = []
        current = self.head.nextNode
        while current != self.tail:
	        resultKeys.append(current.key)
	        current = current.nextNode
        return resultKeys

    def remove_key(self, key):
        if key not in self.cache_map:
            return False
        node = self.cache_map[key]
        self._remove_node(node)
        del self.cache_map[key]
        return True

    def __len__(self):
        return len(self.cache_map)

    def __contains__(self, key):
        return key in self.cache_map

File: vibecheck-core/tests/fixtures/lru_cache/copilot.go
package lrucache

import "container/list"

const (
	defaultCap = iota + 1
	smallCap
	mediumCap
	largeCap
)

type entry struct {
	key   string
	value interface{}
}

type LRUCache struct {
	capacity  int
	cacheMap  map[string]*list.Element
	evictList *list.List
}

func NewCache(capacity int) *LRUCache {
	return &LRUCache{
		capacity:  capacity,
		cacheMap:  make(map[string]*list.Element),
		evictList: list.New(),
	}
}

func (c *LRUCache) Get(key string) (interface{}, bool) {
	elem, ok := c.cacheMap[key]
	if !ok {
		return nil, false
	}
	c.evictList.MoveToFront(elem)
	return elem.Value.(*entry).value, true
}

func (c *LRUCache) Put(key string, value interface{}) {
	if elem, ok := c.cacheMap[key]; ok {
		c.evictList.MoveToFront(elem)
		elem.Value.(*entry).value = value
		return
	}
	if c.evictList.Len() >= c.capacity {
		c.evict_oldest()
	}
	newEntry := &entry{key: key, value: value}
	elem := c.evictList.PushFront(newEntry)
	c.cacheMap[key] = elem
}

func (c *LRUCache) evict_oldest() {
	oldest := c.evictList.Back()
	if oldest == nil {
		return
	}
	c.evictList.Remove(oldest)
	oldEntry := oldest.Value.(*entry)
	delete(c.cacheMap, oldEntry.key)
}

func (c *LRUCache) Remove(key string) bool {
	elem, ok := c.cacheMap[key]
	if !ok {
		return false
	}
	c.evictList.Remove(elem)
	delete(c.cacheMap, key)
	return true
}

func (c *LRUCache) Len() int { return c.evictList.Len() }

func (c *LRUCache) Contains(key string) bool {
	_, ok := c.cacheMap[key]
	return ok
}

func (c *LRUCache) peek_value(key string) (interface{}, bool) {
	elem, ok := c.cacheMap[key]
	if !ok {
		return nil, false
	}
	return elem.Value.(*entry).value, true
}

func (c *LRUCache) Clear() {
	c.cacheMap = make(map[string]*list.Element)
	c.evictList.Init()
}

func (c *LRUCache) get_keys() []string {
	resultKeys := make([]string, 0, c.evictList.Len())
	for elem := c.evictList.Front(); elem != nil; elem = elem.Next() {
	    resultKeys = append(resultKeys, elem.Value.(*entry).key)
	}
	return resultKeys
}

func (c *LRUCache) GetOldest() (string, interface{}, bool) {
	oldest := c.evictList.Back()
	if oldest == nil {
		return "", nil, false
	}
	oldEntry := oldest.Value.(*entry)
	return oldEntry.key, oldEntry.value, true
}
